### DISCUSSION TITLE
Setting SVG viewbox

### DISCUSSION THREAD

**Discussion by frankvp11** [zauberzeug/nicegui#3467] [Ideas / Feature Requests]
I realize that you are able to set the SVG viewbox when there is no source defined in an interactive image.
However, I would like to still be able to set the dimensions of the SVG viewbox while there is a image source. The motivation for this is to display annotations from ML models ontop of the image. If the native image size if 640x480, but the webpage resizes it to some other size, the SVG will also be resized, placing the annotations in the wrong spot. However, setting the SVG viewbox size to 640x480 (hardcoded) would fix this issue. The reason that I don't want to just scale the annotations is because this is more prone to errors, requires more updates, and wouldn't get updated it the screen size changed. 
I have yet to test, (but I am pretty sure) that changing these lines to remove the `!this.src` would do it. 
https://github.com/zauberzeug/nicegui/blob/fca2b5503ceefa2125b2c297ac842fdad2b11953/nicegui/elements/interactive_image.js#L77
however it could also require altering the onImageLoaded function.

**Comment by rodja** [comment]
> If the native image size if 640x480, but the webpage resizes it to some other size, the SVG will also be resized, placing the annotations in the wrong spot.

Do you have a [minimum reproducible code example](https://github.com/zauberzeug/nicegui/wiki/FAQs#why-is-it-important-to-provide-a-minimal-reproducible-example)? The interactive image should take care of the scaling already and set the viewbox in a way that you can simply use the original image coordinates. See https://nicegui.io/documentation/interactive_image#svg_events for a demo where the parameters of the `rect` tags are using screen coordinates instead of the ones from the browser.

**Reply by frankvp11** [reply]
Hmm that's really interesting. I wonder then why it was placing them in the right spot but locally it wasn't. I'll have to look more into my specific use case to see if I can get a MRE out of it
